server/models/Collective.js

```diff
@@ -1265,134 +1265,133 @@ export default function(Sequelize, DataTypes) {
-              },
-              sequelizeParams,
-            ),
-            urlPath: this.getUrlPath(),
-          }).then(({ order, urlPath, memberCollective }) => {
-            const data = {
-              collective: { ...this.minimal, urlPath },
-              member: {
-                ...member.info,
-                memberCollective: memberCollective.activity,
-              },
-              order: order && {
-                ...order.activity,
-                tier: order.Tier && order.Tier.minimal,
-                subscription: {
-                  interval: order.Subscription && order.Subscription.interval,
-                },
-              },
-            };
-            return models.Activity.create(
-              {
-                CollectiveId: this.id,
-                type: activities.COLLECTIVE_MEMBER_CREATED,
-                data,
-              },
-              sequelizeParams,
-            );
-          });
+      },
+    };
 
-        case roles.MEMBER:
-        case roles.ADMIN:
-          // We don't notify if the new member is the logged in user
-          if (get(remoteUser, 'collective.id') === get(memberUser, 'collective.id')) {
-            return member;
-          }
-          // We only send the notification for new member for role MEMBER and ADMIN
-          return emailLib.send(
-            `${this.type}.newmember`.toLowerCase(),
-            memberUser.email,
-            {
-              remoteUser: {
-                email: remoteUser.email,
-                collective: pick(remoteUser.collective, ['slug', 'name', 'image']),
-              },
-              role: role.toLowerCase(),
-              isAdmin: role === roles.ADMIN,
-              collective: {
-                slug: this.slug,
-                name: this.name,
-                type: this.type.toLowerCase(),
-              },
-              recipient: {
-                collective: memberUser.collective.activity,
-              },
-              loginLink: `${config.host.website}/signin?next=/${memberUser.collective.slug}/edit`,
-            },
-            { cc: remoteUser.email },
-          );
-        default:
-          return member;
-      }
-    });
+    return models.Activity.create(
+      { CollectiveId: this.id, type: activities.COLLECTIVE_MEMBER_CREATED, data },
+      sequelizeParams,
+    );
+  };
+
+  Collective.prototype.sendNewMemberEmail = async function(user, role, member, sequelizeParams) {
+    const remoteUser = await models.User.findByPk(
+      member.CreatedByUserId,
+      { include: [{ model: models.Collective, as: 'collective' }] },
+      sequelizeParams,
+    );
+
+    const memberUser = await models.User.findByPk(
+      user.id,
+      { include: [{ model: models.Collective, as: 'collective' }] },
+      sequelizeParams,
+    );
+
+    // We don't notify if the new member is the logged in user
+    if (get(remoteUser, 'collective.id') === get(memberUser, 'collective.id')) {
+      return;
```

Comment:
I don't understand this change

Reply:
We're in `sendNewMemberEmail` not in `addUserWithRole` so it doesn't have to return anything.